### PR TITLE
move cache outside of attention

### PR DESCRIPTION
**Pull request by svtdanny**

### CODE REVIEW

File: wrappers/python/nntile/nntile_core.cc [hunk 1]
```diff
@@ -307,8 +307,7 @@ void tensor_from_array(const tensor::Tensor<T> &tensor,
 #endif // STARPU_SIMGRID
         tile_local.release();
     }
-    tensor::scatter<T>(tmp, tensor);
-    tmp.unregister();
+    tensor::scatter_async<T>(tmp, tensor);
```
Comment (Muxas) [inline wrappers/python/nntile/nntile_core.cc:310]: On the line 368 please make `gather` also async. It will remove a burden of waiting ALL THE TASKS to finish from the `tensor_to_array`. Only `gather_async` will be awaited by the following `acquire` call.

File: wrappers/python/tests/layer/test_attention.py [hunk 2]
```diff
@@ -292,7 +292,7 @@ def test_dynamic(
     )
     attn.init_randn_async()
 
-    out_dynamic_actual = attn.forward_dynamic(inp_tm)
+    out_dynamic_actual, _ = attn.forward_dynamic(inp_tm)
```
Comment (Muxas) [inline wrappers/python/tests/layer/test_attention.py:295]: Could you please add a test, that running a prefill with n-2 tokens and 2 subsequent decodes generates nearly the same tensor of logits as if a single decode with n tokens?

**Review by Muxas — CHANGES_REQUESTED** [review]

**Review by Muxas — APPROVED** [review]
